### PR TITLE
User profile

### PR DESCRIPTION
**Pull request by xbaro**
Added new fields to the user profile. This information will be useful for organizers in order to know the profile of the participants and also for participants in order to find other participants to create teams in the competitions.

The added information is:
- ORCID number
- Biography
- Linkedin
- Webpage

In addition, a flag to allow users to decide whether to make the profile public (visible to other users or not) has been added to the profile.

Finally, in order to avoid data protection issues, a new flag at application settings level has been added in order to allow the administrators of a Codalab instance to decide whether to use the extended profile for users or only the basic one.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,12 @@ To see Codalab Competition's in action, visit [www.competitions.codalab.org](htt
 The CodaLab community forum is hosted on Google Groups.
 - [CodaLabDev Google Groups Forum](https://groups.google.com/forum/#!forum/codalabdev)
 
+## Tests
+
+To run tests:
+
+```python manage.py test```
```
Comment (ckcollab) [inline README.md]: Same as other review, maybe just check this file out from develop?

File: codalab/apps/authenz/models.py [hunk 2]
```diff
@@ -1,5 +1,25 @@
+import os
 from django.db import models
 from django.contrib.auth import models as auth_models
+from django.utils.functional import cached_property
+from django.conf import settings
+from django.core.files.storage import get_storage_class
+
+## Needed for computation service handling
```
Comment (ckcollab) [inline codalab/apps/authenz/models.py]: We should pull this behavior out from `web/models.py` and put it in base.py, so we can just do `settings.BundleStorage`.

```
StorageClass = get_storage_class(settings.DEFAULT_FILE_STORAGE)

if hasattr(settings, 'USE_AWS') and settings.USE_AWS:
    BundleStorage = StorageClass(bucket=settings.AWS_STORAGE_PRIVATE_BUCKET_NAME)
    PublicStorage = StorageClass(bucket=settings.AWS_STORAGE_BUCKET_NAME)
elif hasattr(settings, 'BUNDLE_AZURE_ACCOUNT_NAME'):
    BundleStorage = StorageClass(account_name=settings.BUNDLE_AZURE_ACCOUNT_NAME,
                                 account_key=settings.BUNDLE_AZURE_ACCOUNT_KEY,
                                 azure_container=settings.BUNDLE_AZURE_CONTAINER)

    PublicStorage = StorageClass(account_name=settings.AZURE_ACCOUNT_NAME,
                                 account_key=settings.AZURE_ACCOUNT_KEY,
                                 azure_container=settings.AZURE_CONTAINER)
else:
    # No storage provided, like in a test, let's just do something basic
    BundleStorage = StorageClass()
    PublicStorage = StorageClass()
```

File: codalab/apps/authenz/models.py [hunk 3]
```diff
@@ -25,3 +53,25 @@ class ClUser(auth_models.AbstractUser):
     bibtex = models.TextField(null=True, blank=True)
 
     contact_email = models.EmailField(null=True, blank=True)
+
+    # Allow the user to decide if the profile is public or not
+    public_profile = models.BooleanField(default=False)
+
+    def update_filename(instance, filename):
+        path = "user_photo"
+        format = instance.username + instance.file_extension
```
Comment (ckcollab) [inline codalab/apps/authenz/models.py]: use `"{}{}".format(instance.username, instance.file_extension)` instead

File: codalab/apps/web/forms.py [hunk 4]
```diff
@@ -224,10 +225,34 @@ class Meta:
...
         }
+
+        def __init__(self, *args, **kwargs):
+            super(UserSettingsForm, self).__init__(*args, **kwargs)
+
+            self.fields["extended_profile"] = not settings.USE_EXTENDED_PROFILE
+
+        def clean_image(self):
+            if 'image' in self.changed_data:
+                file, ext = os.path.splitext(self.files['image'].name)
```
Comment (ckcollab) [inline codalab/apps/web/forms.py]: dont override `file` builtin

File: codalab/codalab/settings/base.py [hunk 5]
```diff
@@ -68,6 +68,10 @@ class Base(Settings):
 
     MANAGERS = ADMINS
 
+    # If enabled, the user profile contains all defined information (Organization, biography, image, ...),
+    # Otherwise, only basic information is collected for users, avoiding data protection issues.
+    USE_EXTENDED_PROFILE = True
```
Comment (ckcollab) [inline codalab/codalab/settings/base.py]: nice 👍

File: codalab/codalabtools/deploy/fabfile.py [hunk 6]
```diff
@@ -125,7 +125,12 @@ def provision_compute_workers_packages():
     """
     Installs required software packages on a newly provisioned compute worker machine.
     """
-    packages = ('python-crypto libpcre3-dev libpng12-dev libjpeg-dev libmysqlclient-dev uwsgi-plugin-python libsm6 openjdk-7-jre upstart upstart-sysv unixodbc unixodbc-dev')
+    packages = ('python-crypto libpcre3-dev libpng12-dev libjpeg-dev libmysqlclient-dev uwsgi-plugin-python libsm6 openjdk-7-jre unixodbc unixodbc-dev')
+
+    ubuntu_version = run('cat /etc/issue')
```
Comment (ckcollab) [inline codalab/codalabtools/deploy/fabfile.py]: hmm, again in this PR, we have some extra stuff. We're going to be using docker, so I think we can ignore this for now? I'll probably be merging docker into develop once I get tests passing here soon.

File: codalab/apps/authenz/models.py (hunk 2 above)
Comment (ckcollab) [inline codalab/apps/authenz/models.py]: I did this in the docker branch, it's available from `from apps.web.utils import PublicStorage, BundleStorage`

**Comment by xbaro** [comment]
I've merged the develop branch to this one in order to be able to use the Storage on the new format. Also changed what you commented.

**Comment by ckcollab** [comment]
Closing this for now until we need this feature

**Pull request closed without merging**